### PR TITLE
Matrix permanent

### PR DESCRIPTION
**Pull request by jrexmo**
#### Brief description of what is fixed or changed
Now includes a method for computing the permanent of a matrix via sums over products of elements (e.g. the classical definition).

#### Other comments
Could potentially be made faster, this is a todo.

#### Release Notes

* matrices
  * Added a new method for computing the permanent of a matrix

### CODE REVIEW

File: sympy/matrices/determinant.py [hunk 1]
```diff
@@ -828,3 +828,47 @@ def _minor_submatrix(M, i, j):
...
+
+    # Computing permanent by summing over all permuatations on (0, ... n-1)
+    # TODO: find a faster way to do this, maybe a sparse method, maybe Gaussian Elim
+    for perm in perm_list:
+
+        product = 1
+
+        for i in range(n):
+
+            product = product*M[i, perm[i]]
```
Comment (sylee957) [inline sympy/matrices/determinant.py]: I think that you should clean up some redundant spacings in the code

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -974,11 +975,6 @@ def _handle_creation_inputs(cls, *args, **kwargs):
             elif isinstance(args[0], Basic) and args[0].is_Matrix:
                 return args[0].rows, args[0].cols, args[0].as_explicit()._mat
 
-            elif isinstance(args[0], mp.matrix):
-                M = args[0]
-                flat_list = [cls._sympify(x) for x in M]
-                return M.rows, M.cols, flat_list
```
Comment (sylee957) [inline sympy/matrices/matrices.py]: Why do you delete this?

**Comment by sylee957** [comment]
Please add some tests in test_determinant.py.
A few tests using matrix with symbolic entries would suffice.

File: sympy/matrices/matrices.py (hunk 2 above)
Comment (jrexmo) [reply sympy/matrices/matrices.py]: I am not sure, I will revert.

File: sympy/matrices/determinant.py (hunk 1 above)
Comment (jrexmo) [reply sympy/matrices/determinant.py]: Got it, I will take care of that.

**Pull request closed without merging**